test/cypress/integration/example09.spec.js

```diff
@@ -832,7 +832,7 @@ describe('Example 09 - OData Grid', { retries: 1 }, () => {
 
       cy.get('[data-test=odata-query-result]')
         .should(($span) => {
-          expect($span.text()).to.eq(`$top=10&$orderby=Name desc&$select=name,gender,company,id&$expand=category($select=name)`);
+          expect($span.text()).to.eq(`$top=10&$orderby=Name desc&$select=name,gender,company&$expand=category($select=name)`);
```

Comment:
I see you removed the `rowId` and `id` is no longer part of the `$select`, but will you still get a unique `id` in your dataset? Because you know that SlickGrid will throw an error if you don't have an `id` defined in your data

Reply:
Yes, I mentioned that here https://github.com/ghiscoding/slickgrid-universal/pull/588#discussion_r774921405 

So the user (me) needs to add a column with a field `id`.  That column can be hidden of course if he doesn't want to have it in the UI. 

But it does not have to be `id` and can be anything like `customerId`, or multiple fields and if so in his `backendServiceApi.postProcess` the user can change what the backend send and make sure each item gets an `id`.

Reply:
hmm in GraphQL Service I add it automatically, so I still wonder if it wouldn't be better to do it in here as well, see below. 
If you still think it's unnecessary then let me know and I'll merge as it is. 

Thanks

https://github.com/ghiscoding/slickgrid-universal/blob/37c6ef68e802e634f2ed2ef70800f753a4b2d8eb/packages/graphql/src/services/graphql.service.ts#L72

https://github.com/ghiscoding/slickgrid-universal/blob/37c6ef68e802e634f2ed2ef70800f753a4b2d8eb/packages/graphql/src/services/graphql.service.ts#L109-L113

Reply:
Aha. I didn't know about `gridOptions.datasetIdPropertyName`. Then it's probably smart to align the odata/graphql implementations. 

I'm not sure how to best forward the `grid` or `gridOptions` or perhaps just `datasetIdPropertyName` to `OdataQueryBuilderService`?  I'm leaning towards adding `init(grid) { ... }` to `OdataQueryBuilderService`. Do you have some guidance here? :-)

Reply:
yeah `gridOptions.datasetIdPropertyName` is what you use when you want something else than `id`, I'm not sure if I have it anywhere in the Wikis but it's in a few examples. The `OdataQueryBuilderService` wasn't meant to know about the grid, all it knows and has access is the `odataOption`, it's probably better to keep it that way, so I think you can just add a new option into the `odataOption` and you can let the user provide it and/or use `datasetIdPropertyName` when not. Below I'm using `entityId` but it could be something else `dataId`, `datasetIdProp`, ... most people won't fill that property and it will automatically fall to `gridOptions.datasetIdPropertyName` (that grid option is initialized in global options)

https://github.com/ghiscoding/slickgrid-universal/blob/37c6ef68e802e634f2ed2ef70800f753a4b2d8eb/packages/common/src/global-grid-options.ts#L101

```ts
init(serviceOptions?: Partial<OdataOption>, pagination?: Pagination, grid?: SlickGrid, sharedService?: SharedService): void {
    this._grid = grid;
    const mergedOptions = { ...this.defaultOptions, ...serviceOptions };
    this._datasetIdPropName = serviceOptions?.entityId ?? this._gridOptions.datasetIdPropertyName ?? 'id';
    // ...
}
```

Reply:
BTW, have you already upgraded to Angular-Slickgrid 4.0.0 that I released 2 weeks ago? Because that would fall under that version only

Reply:
OK. I now forward the `_gridOptions.datasetIdPropertyName` to the `OdataQueryBuilderService`. I didn't extend the `odataOptions` because that doesn't add much I think and it can always be done later.

And yes, we're on v4!

Can you merge? :-)

Reply:
yes I can, awesome work 👍